commit 0a93751d75e02a1471ba4aa5be342e91d603da1a (refs/changes/45/5645/1)
Author: Robert Mustacchi <rm@joyent.com>
Date:   2019-02-27T01:51:01+00:00 (7 months ago)
    
    OS-7619 gcc7 needs to use external names for retpoline thunks

diff --git a/gcc7/Patches/thunk.patch b/gcc7/Patches/thunk.patch
new file mode 100644
index 0000000..0e1e946
--- /dev/null
+++ b/gcc7/Patches/thunk.patch
@@ -0,0 +1,24 @@
+From 3333fee43a68365bb845fa8893782822832200c8 Mon Sep 17 00:00:00 2001
+From: Richard Lowe <richlowe@richlowe.net>
+Date: Wed, 7 Feb 2018 02:13:42 +0000
+Subject: [PATCH] i386: use the new-style retpoline thunk names for external
+ thunks, because nothing else will work
+
+---
+ gcc/config/i386/i386.c | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/gcc/config/i386/i386.c b/gcc/config/i386/i386.c
+index 84a5f7c219ca..c6566d61b0d6 100644
+--- a/gcc/config/i386/i386.c
++++ b/gcc/config/i386/i386.c
+@@ -12067,7 +12067,8 @@ indirect_thunk_name (char name[32], int regno, bool need_bnd_p,
+   if (regno >= 0 && ret_p)
+     gcc_unreachable ();
+ 
+-  if (USE_HIDDEN_LINKONCE)
++  if (USE_HIDDEN_LINKONCE ||
++      (cfun->machine->indirect_branch_type == indirect_branch_thunk_extern))
+     {
+       const char *bnd = need_bnd_p ? "_bnd" : "";
+       if (regno >= 0)
